compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/transform/SimpleDoFnTransform.java

```diff
@@ -0,0 +1,152 @@
+  private final TupleTag<OutputT> mainOutputTag;
+  private final List<TupleTag<?>> additionalOutputTags;
+  private final Collection<PCollectionView<?>> sideInputs;
+  private final WindowingStrategy<?, ?> windowingStrategy;
+  private final DoFn<InputT, OutputT> doFn;
+  private final SerializablePipelineOptions serializedOptions;
+  private transient DoFnRunner<InputT, OutputT> doFnRunner;
+  private transient SideInputReader sideInputReader;
+  private transient DoFnInvoker<InputT, OutputT> doFnInvoker;
+  private final Coder<InputT> inputCoder;
+  private final Map<TupleTag<?>, Coder<?>> outputCoders;
+
+  /**
+   * DoTransform Constructor.
+   *
+   * @param doFn    doFn.
+   * @param options Pipeline options.
+   */
+  public SimpleDoFnTransform(final DoFn<InputT, OutputT> doFn,
+                             final Coder<InputT> inputCoder,
+                             final Map<TupleTag<?>, Coder<?>> outputCoders,
+                             final TupleTag<OutputT> mainOutputTag,
+                             final List<TupleTag<?>> additionalOutputTags,
+                             final WindowingStrategy<?, ?> windowingStrategy,
+                             final Collection<PCollectionView<?>> sideInputs,
+                             final PipelineOptions options) {
+    this.doFn = doFn;
+    this.inputCoder = inputCoder;
+    this.outputCoders = outputCoders;
+    this.mainOutputTag = mainOutputTag;
+    this.additionalOutputTags = additionalOutputTags;
+    this.sideInputs = sideInputs;
+    this.serializedOptions = new SerializablePipelineOptions(options);
+    this.windowingStrategy = windowingStrategy;
+  }
+
+  @Override
+  public void prepare(final Context context, final OutputCollector<WindowedValue<OutputT>> oc) {
+    // deserialize pipeline option
+    final NemoPipelineOptions options = serializedOptions.get().as(NemoPipelineOptions.class);
+
+    this.outputCollector = oc;
+
+    // create output manager
+    final DoFnRunners.OutputManager outputManager = new DefaultOutputManager<>(
+      outputCollector, context, mainOutputTag);
+
+    // create side input reader
+    sideInputReader = NullSideInputReader.of(sideInputs);
+    if (!sideInputs.isEmpty()) {
+      sideInputReader = new BroadcastGlobalValueSideInputReader(context, sideInputs);
+    }
+
+    // create step context
+    // this transform does not support state and timer.
+    final StepContext stepContext = new StepContext() {
+      @Override
+      public StateInternals stateInternals() {
+        throw new UnsupportedOperationException("Not support stateInternals in SimpleDoFnTransform");
+      }
+
+      @Override
+      public TimerInternals timerInternals() {
+        throw new UnsupportedOperationException("Not support timerInternals in SimpleDoFnTransform");
+      }
+    };
+
+    // invoker
+    doFnInvoker = DoFnInvokers.invokerFor(doFn);
+    doFnInvoker.invokeSetup();
+
+    // runner
+    doFnRunner = DoFnRunners.simpleRunner(
```

Comment:
This is much cleaner than the previous code!
Can you add a friendly note that the `DoFnRunners.simpleRunner` takes care of all the hard stuff of running the DoFn (sideinput, windowing, multi-output, etc), and that this approach is the standard used by most of the Beam runners?